compute/cker/src/train/AvgPool.test.cc

```diff
@@ -0,0 +1,286 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cker/eigen/Utils.h>
+#include <cker/operation/AveragePool.h>
+#include <cker/train/operation/AvgPool.h>
+#include <cker/Shape.h>
+
+#include <gtest/gtest.h>
+#include <vector>
+
+namespace
+{
+using namespace nnfw::cker;
+
+template <typename T> class AvgPoolOpVerifier
+{
+private:
+  const PoolParams _op_params;
+  const Shape _in_shape;
+  const Shape _out_shape;
+
+public:
+  AvgPoolOpVerifier(const nnfw::cker::PoolParams &op_params, const Shape &in_shape,
+                    const Shape &out_shape)
+    : _op_params(op_params), _in_shape(in_shape), _out_shape(out_shape)
+  {
+  }
+
+public:
+  void verifyForward(const std::vector<T> input, const std::vector<T> expected_output,
+                     bool expect_eq = true)
+  {
+    assert(input.size() == _in_shape.FlatSize());
+    assert(expected_output.size() == _out_shape.FlatSize());
+
+    std::vector<T> cacluated_output(_out_shape.FlatSize());
+    nnfw::cker::train::AvgPool2D(_op_params, _in_shape, input.data(), _out_shape,
+                                 cacluated_output.data());
+
+    if (expect_eq)
+      for (size_t i = 0; i < expected_output.size(); i++)
+      {
+        EXPECT_FLOAT_EQ(expected_output[i], cacluated_output[i]);
+      }
+    else
+      EXPECT_NE(expected_output, cacluated_output);
+  }
+
+  void verifyBackward(const std::vector<T> incoming_data, const std::vector<T> expected_grad_data,
+                      bool expect_eq = true)
+  {
+    assert(incoming_data.size() == _out_shape.FlatSize());
+    assert(expected_grad_data.size() == _in_shape.FlatSize());
+
+    std::vector<T> calcuated_grad(_in_shape.FlatSize());
+    nnfw::cker::train::AvgPool2DGrad(_op_params, _out_shape, incoming_data.data(), _in_shape,
+                                     calcuated_grad.data());
+
+    if (expect_eq)
+    {
+      for (size_t i = 0; i < expected_grad_data.size(); i++)
+      {
+        EXPECT_FLOAT_EQ(expected_grad_data[i], calcuated_grad[i]);
+      }
+    }
+
+    else
+      EXPECT_NE(expected_grad_data, calcuated_grad);
+  }
+};
+
+} // namespace
+
+TEST(CKer_Operation, AvgPool2D)
+{
+  // Depth 1 case
+  {
+    nnfw::cker::PoolParams op_param;
+    {
+      op_param.stride_height = 1;
+      op_param.stride_width = 1;
+      op_param.filter_height = 2;
+      op_param.filter_width = 2;
+      op_param.padding_values.height = 0;
+      op_param.padding_values.width = 0;
+    }
+    nnfw::cker::Shape in = {1, 3, 3, 1};
+    nnfw::cker::Shape out = {1, 2, 2, 1};
+
+    AvgPoolOpVerifier<float> verifier(op_param, in, out);
+
+    /**
+     *  input(index) :                         output(arg-count):
```

Comment:
```suggestion
     *  input(index) :                         output(arg-count):
```

May I ask what is the meaning of `arg_count`? 
I also wonder that line150 used `index` instead of `arg-count`. Is there a reason for this difference?

Reply:
Ah, If you followed the `maxpool2d` notation, `maxpool2d` expressed the index - because one output of  `maxpool2d` is arg-index. 

For avergepool2d, index isn't important, I guess you don't need to mark the index here.  
If you have another intention about `arg-count`(or `index`)  plz let me know.